File: solutions/lua/leap/1/leap.lua
local leap_year = function(number)
  local divisibleBy4 = number % 4 == 0
  local divisibleBy100 = number % 100 == 0
  local divisibleBy400 = number % 400 == 0

  return divisibleBy400 or divisibleBy4 and not divisibleBy100
end

return leap_year

File: solutions/ruby/leap/2/leap.rb
# frozen_string_literal: true

# Handle resistors resistance value based on their color
module Year
  module_function

  # @param year [Integer]
  def leap?(year)
    divisible_by4 = (year % 4).zero?
    divisible_by100 = (year % 100).zero?
    divisible_by400 = (year % 400).zero?

    divisible_by400 || divisible_by4 && !divisible_by100
  end
end
